common/xbps-src/shutils/lint_commits.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+lint_commits() {
+	local base="$1"
+	local status=0
+	local cmt
+	if [ -n "$base" ]; then
+		base=$("$XBPS_GIT_CMD" rev-parse --verify "$base") ||
+			return
+	elif base=$("$XBPS_GIT_CMD" rev-parse --verify FETCH_HEAD 2>/dev/null); then
+		:
+	elif base=$("$XBPS_GIT_CMD" rev-parse --verify ORIG_HEAD 2>/dev/null); then
+		:
+	else
+		echo base commit not found >&2
+		exit 1
+	fi
+
+	for cmt in $("$XBPS_GIT_CMD" rev-list --reverse --abbrev-commit "$base"..HEAD)
+	do
+		"$XBPS_GIT_CMD" cat-file commit "$cmt" |
+		awk -vC="$cmt" '
+		/^$/ && !msg { msg = 1; next }
+		!msg { next }
+		!/ / && (length > 80) { print C ": long line: " $0; exit 1 }
+		!subject {
+			if (length > 50) { print C ": subject is a bit long" }
+			# Below check is too noisy?
+			# if (!($0 ~ "^New package:" || $0 ~ ".*: update to")) {
+			#	print C ": not new package/update/removal?"
+			# }
```

Comment:
It would be noisy for test fixes and xbps-src changes, not sure it's worth it but I don't think it'd be that noisy either.

Reply:
I think we should discuss that later. I really want to have some lint for blank line and long lines soon.